src/Microsoft.AspNet.OData.Shared/Builder/ActionConfiguration.cs

```diff
@@ -233,6 +233,37 @@ public ActionConfiguration SetBindingParameter(string name, IEdmTypeConfiguratio
             return this;
         }
 
+        /// <summary>
+        /// Adds subtypes to the list of derived type constraints for the return type.
+        /// </summary>
+        /// <param name="subtypes">The subtypes for which the constraint needs to be added.</param>
+        /// <returns>Updated configuration object.</returns>
+        public ActionConfiguration HasDerivedTypeConstraints(params Type[] subtypes)
```

Comment:
>HasDerivedTypeConstraints [](start = 35, length = 25)

I can't understand which derived constraint happens if customer call this method.

It's for the Binding type constraint, or return type constraint.

Be caution, return type and binding type could be the same type.

Reply:
It is for the return type. Binding type is more of a parameter configuration than function configuration.

Reply:
See my changes: [https://github.com/OData/WebApi/blob/9c9257fd05783ae5b8ed427b9eef1848f0fa48c5/src/Microsoft.AspNet.OData.Shared/Builder/OperationConfiguration.cs](https://github.com/OData/WebApi/blob/9c9257fd05783ae5b8ed427b9eef1848f0fa48c5/src/Microsoft.AspNet.OData.Shared/Builder/OperationConfiguration.cs)

---
In reply to: [394514464](https://github.com/OData/WebApi/pull/1977#discussion_r394514464) [](ancestors = 394514464)